lib/galaxy/webapps/galaxy/api/cloud.py

```diff
@@ -0,0 +1,121 @@
+"""
+API operations on Cloud-based storages, such as Amazon Simple Storage Service (S3).
+"""
+
+import logging
+
+from galaxy import exceptions
+from galaxy.exceptions import ActionInputError
+from galaxy.managers import (
+    cloud,
+    datasets
+)
+from galaxy.web import _future_expose_api as expose_api
+from galaxy.web.base.controller import BaseAPIController
+
+log = logging.getLogger(__name__)
+
+
+class CloudController(BaseAPIController):
+    """
+    RESTfull controller for interaction with Amazon S3.
+    """
+
+    def __init__(self, app):
+        super(CloudController, self).__init__(app)
+        self.cloud_manager = cloud.CloudManager(app)
+        self.datasets_serializer = datasets.DatasetSerializer(app)
+
+    @expose_api
+    def index(self, trans, **kwargs):
+        """
+        * GET /api/cloud
+            Lists cloud-based containers (e.g., S3 bucket, Azure blob) user has defined.
```

Comment:
How about settling on term `bucket` instead of `container` throughout? It's the term used in CloudBridge and avoids some of the potential confusion with software containers.